tests/Aspire.Hosting.Containers.Tests/FromDockerfileTests.cs

```diff
@@ -0,0 +1,358 @@
+               .WithDockerfile(tempContextPath, tempDockerfilePath);
+
+        using var app = builder.Build();
+        await app.StartAsync();
+
+        using var client = app.CreateHttpClient("testcontainer", "http");
+        var message = await client.GetStringAsync("/aspire.html"); // Proves the container built, ran, and contains customizations!
+
+        Assert.Equal($"{DefaultMessage}\n", message);
+
+        var kubernetes = app.Services.GetRequiredService<IKubernetesService>();
+        var containers = await kubernetes.ListAsync<Container>();
+
+        var container = Assert.Single<Container>(containers);
+        Assert.Equal(tempContextPath, container!.Spec!.Build!.Context);
+        Assert.Equal(tempDockerfilePath, container!.Spec!.Build!.Dockerfile);
+
+        await app.StopAsync();
+    }
+
+    [Fact]
+    public async Task WithDockerfileResultsInBuildAttributeBeingAddedToManifest()
+    {
+        var (tempContextPath, tempDockerfilePath) = await CreateTemporaryDockerfileAsync();
+        var manifestOutputPath = Path.Combine(tempContextPath, "aspire-manifest.json");
+        var builder = DistributedApplication.CreateBuilder(new DistributedApplicationOptions
+        {
+            Args = ["--publisher", "manifest", "--output-path", manifestOutputPath],
+        });
+
+        var parameter = builder.AddParameter("message");
+        builder.Configuration["Parameters:message"] = "hello";
+
+        var container = builder.AddContainer("testcontainer", "testimage")
+                               .WithHttpEndpoint(targetPort: 80)
+                               .WithDockerfile(tempContextPath, tempDockerfilePath, "runner")
+                               .WithBuildArg("MESSAGE", parameter)
+                               .WithBuildArg("stringParam", "a string")
+                               .WithBuildArg("intParam", 42);
+
+        var manifest = await ManifestUtils.GetManifest(container.Resource, manifestDirectory: tempContextPath);
+        var expectedManifest = $$$$"""
+            {
+              "type": "container.v0",
+              "image": "testimage:latest",
+              "build": {
+                "context": ".",
+                "dockerfile": "Dockerfile",
+                "stage": "runner",
+                "args": {
+                  "MESSAGE": "{message.value}",
+                  "stringParam": "a string",
+                  "intParam": "42"
+                }
+              },
+              "bindings": {
+                "http": {
+                  "scheme": "http",
+                  "protocol": "tcp",
+                  "transport": "http",
+                  "targetPort": 80
+                }
+              }
+            }
+            """;
+        Assert.Equal(expectedManifest, manifest.ToString());
+    }
+
+    [Fact]
+    public async Task WithDockerfileWithParameterLaunchesContainerSuccessfully()
+    {
+        if (!IsDockerAvailable())
+        {
+            return;
+        }
```

Comment:
You should be able to use `[RequiresDocker]` with the test. Though it assumes that `docker` is available unless running on Windows+Helix. And the reason for that is to make sure that tests don't get silently skipped on CI if docker isn't failing somehow.